### DISCUSSION TITLE
Export CSV of Branch / Endpoints

### DISCUSSION THREAD

**Discussion by GriEri** [JacobBumgarner/VesselVio#48] [Ideas]
Is it possible to export a list of all the coordinates of branchpoints / endpoints? Thank you for any consideration or tips.
Is there an array which contains the XYZ positions of branching points I can export? what would be its name?

**Comment by JacobBumgarner** [comment]
Hi @GriEri! This feature doesn't exist yet, but it’s a good idea and will be simple to implement.

In your mind, what information would you want provided in the branch/endpoint files? Just coordinates?

**Reply by GriEri** [reply]
Just the coordinates would be fine. I'm interested in density of branching points and finding areas with high/low density, but all analysis can be done separately if a CSV of XYZ coordinates can be exported.

**Comment by JacobBumgarner** [comment]
To the second part of your question - you will isolate that information from the graph object.

Each graph vertex has an attribute ‘v_coords’. 

You can isolate the endpoints and branch points into a vertex sequence as such:

```python
branch_points = g.vs.select(_degree_gt=2)
endpoints = g.vs.select(_degree=1)
```

You can then use those vertex sequences to create subgraphs to to extract a list of the coordinates, degrees, etc. This is all done with igraph.

This will be an easier one to implement, but I’m caught up in the middle of a codebase cleanup and refactoring grind, so it might be a few days before I’m able to have it implemented in an effective way.

**Reply by JacobBumgarner** [reply]
If you wouldn’t mind, @GriEri could you open a feature request for this on the Issues page?

**Reply by GriEri** [reply]
Thank you, opening request now

**Comment by JacobBumgarner** [comment]
@GriEri Quick question for you- are you using VesselVio from the terminal or are you using it as a downloaded app? 

I'm almost finished with the rework of the section that will allow me to add this feature. If you're running the app from the terminal, I should be able to get the CSV export feature ready for you by the end of the week.

**Comment by GriEri** [comment]
Hello, I am running it by launching VesselVio.py from within a python environment.
Please don't feel a rush on my behalf, using your advice I made a temporary workaround by adding the following into graph_io.py:

`
BranchPointsGraph = g.subgraph(g.vs.select(_degree_gt=2)) #make subgraph 
BranchPointsGraph.write_graphml(r"SaveLocationAndFileName")  
`

and then I can just open the GRAPHML in R, and extract the coordinates,
While not a polished solution that will work for most, it should work for my research project, and I can wait until a future release.
However, I would be happy to try out any solution you're testing
